proto-lens/src/Data/ProtoLens/Encoding.hs

```diff
@@ -107,12 +107,20 @@ parseAndAddField
               OptionalField f -> do
                   x <- getSimpleVal
                   return $ set f (Just x) msg
-              RepeatedField Unpacked f -> do
-                  x <- getSimpleVal
-                  return $ over f (x:) msg
-              RepeatedField Packed f -> do
-                  xs <- getPackedVals
-                  return $ over f (xs++) msg
+              RepeatedField _ f
+                    -- Parse either a packed or unpacked representation,
+                    -- depending on how it was encoded.
+                    -- Note that if fieldWt is Lengthy (e.g., "string" or
+                    -- message) we should always parse it as unpacked.
+                    | Right Equal <- equalWireTypes name wt fieldWt -> do
```

Comment:
Alternatively (ha), "RepeatedField _ f -> getPackedVals <|> getUnpackedVals <|> fail ..." and keep the "equalWireTypes" calls in the helper functions.